### ISSUE TITLE
Sort hash keys when failures happen

### ISSUE THREAD

**Issue by skington** [DrHyde/perl-modules-Class-Mockable#9]
I got the following output just now, when I made a test deliberately fail to make sure I was actually testing what I thought I was testing:

```
    #   Failed test 'wrong args to mock object 'mock_logger' defined in main::test_user_defined_collation_methods at line 123 of infrastructure.t (expected $VAR1 = [
    #           {
    #             'label' => 'collate_rivers',
    #             'elapsed_time' => '1.189',
    #             'event' => 'report_timer'
    #           }
    #         ];
    # , got $VAR1 = [
    #           {
    #             'elapsed_time' => '0.189',
    #             'label' => 'collate_rivers',
    #             'event' => 'report_timer'
    #           }
    #         ];
    # )'
```

Keys should be reported in the same order for both expected and got, and the simplest way of doing that is to sort them, which should just be a matter of saying `local $Data::Dumper::Sortkeys = 1` in the AUTOLOAD method.

You may as well add Terse while you're there and get rid of the pointless $VAR1.

**Comment by DrHyde** [comment]
It should really use Test::Differences instead of Data::Dumper too.

**Comment by skington** [comment]
Baby steps.

**Comment by DrHyde** [comment]
Fixed in 1.3001
